packages/babel-plugin-proposal-object-rest-spread/test/hasMoreThanOneBinding.test.js

```diff
@@ -0,0 +1,46 @@
+import { parse } from "@babel/parser";
+import hasMoreThanOneBinding from "../lib/hasMoreThanOneBinding";
+
+function getFistObjectPattern(program) {
+  return parse(program, { sourceType: "module" }).program.body[0]
+    .declarations[0].id;
+}
+describe("hasMoreThanOneBinding", function () {
+  it.each([
+    ["const { x: { ...y } } = z();", true],
+    ["let { x4: { ...y4 } } = z();", true],
```

Comment:
It should return `false`, the source contains only one binding `y4`. Also we don't have to add number suffix since every snippet is parsed independently.

Reply:
So I think `hasMoreThanOneBinding` is an invalid name fo the function. The end goal of this is to strictly stop the right hand side being duplicated

I am happy to rename this function or do something else but in the end the main goal is to mitigate the active bug.